app/assets/javascripts/discourse/app/components/d-modal.js

```diff
@@ -124,7 +57,11 @@ export default class DModal extends Component {
   }
 
   @action
-  handleMouseDown(e) {
+  handleMouseUp(e) {
+    if (e.button !== 0) {
+      return; // Non-default mouse button
+    }
```

Comment:
In the old modal implementation, right-clicking on the backdrop would cause it to close. This is pretty annoying, especially in development where you might be trying to 'inspect element'. Taking this opportunity to fix it up.